src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs

```diff
@@ -259,6 +259,7 @@ public IReadOnlyList<string> Tokens
 
         public int Execute(params string[] args)
         {
+            TemplateEngineEventSource.Log.ParseArgsStart();
```

Comment:
note that this is not single time the parsing is done.
Reparsing is done many times afterwards using https://github.com/dotnet/templating/blob/94b1e0839ab49026b3ff654f2b5e5e1a1b4b2c53/src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs#L302